crobat/src/lib.rs

```diff
@@ -48,15 +48,23 @@ impl Crobat {
     }
 
     pub async fn get_subs(
-        &mut self,
+        self,
         host: Arc<String>,
     ) -> Result<impl Stream<Item = std::result::Result<Domain, Status>>> {
-        trace!("querying crobat client for subdomains");
-        let request = tonic::Request::new(QueryRequest {
-            query: host.to_string(),
-        });
-        let stream = self.client.get_subdomains(request).await?.into_inner();
+        match self.client {
+            Some(mut client) => {
+                trace!("querying crobat client for subdomains");
+                let request = tonic::Request::new(QueryRequest {
+                    query: host.to_string(),
+                });
+                let stream = client.get_subdomains(request).await?.into_inner();
 
-        Ok(stream)
+                Ok(stream)
+            }
+            None => Err(Box::new(Status::new(
+                Code::Unavailable,
+                "crobat client failed to initialize connection with server",
```

Comment:
It's a little weird to bubble up the error so late after it happened, but returning an error from a constructor didn't feel right.

Reply:
An alternative might also be just to return a `Result` from the constructor like `File::create` in [std::fs::File](https://doc.rust-lang.org/std/fs/struct.File.html#method.create)
```rust
pub async fn new() -> Result<Self> {
        trace!("building crobat client");
        let addr = "https://crobat-rpc.omnisint.io";
        match  Crobat::build_tls_client(addr).await {
            Ok(conn) => Self { client: CrobatClient::new(conn) },
            Err(e) => Err(CrobatError::ConnectionErr)
    }
}
```
If we went with this approach `new` should probably be renamed to `connect` though, since it seems like a more appropriate name.

Reply:
Ah yeah I guess the pattern isn't so odd afteral! Alright I'll do that